json-logic/schema.ts

```diff
@@ -0,0 +1,146 @@
+// LogicExpression refers to the JSONLogic standard.
+import { LogicExpression } from "./logic";
+
+/*
+* This module defines a schema for questionaire and result calulation logic.
+*/
+
+/**
+ * Type of a question, essentially defines the result type.
+ */
+export enum QuestionType {
+	Select = "Select",
+	Multiselect = "Multiselect",
+	Number = "Number",
+	Boolean = "Boolean",
+	Date = "Date",
+	Text = "Text"
+}
+
+/**
+ * Option for multi-select questions.
+ */
+export interface IOption {
```

Comment:
Should we add an id here? In principle value is enough, but for FHIR QuestionnaireResponse we would have to provide unique ids for options and by defining them explicitly we might avoid errors with duplicate identifiers.

Reply:
I guess in this case we could generate a unique ID by generating something like `questionId.optionValue` and using that as identifier. Not sure if it's necessary to include it here.

Reply:
Well every string that should be translated should have a unique id, so we can translate it more easily

Reply:
Great point!

Reply:
I also think that we can create automatic IDs like Emanuel proposed, if we need to.

Regarding translations: if we use something like PO files (https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html), then we would just need a mapping from original texts to translated texts. No need for IDs then.

Reply:
But with the PO files, you would have to change all references if you choose to update the original text.

Reply:
Once we provide an editor that also has translation capabilities, we should probably think about how to store the texts in our service (probably in a database with some ids).

I still feel that a PO file could be a good format to expose to other systems from our API, since it is straightforward, standardized, and easy to use.

(Once we have a real back-end, we could also offer the possibility to get either the original questionnaire with an external PO file, or the translated questionnaire in a specific language - depending on the endpoint or some parameter.)

Reply:
I'm also not sold on PO files - we JSON everything, we could also use a simple JSON map for providing translations. 

In the beginning, I would also use simple mapping files (maintained and served statically) instead of databases. Moving to a database-backed api is not hard afterwards, as long as we design keep all files well structured.